### ISSUE TITLE
error in find_bottom?

### ISSUE THREAD

**Issue by jeong-tae** [GXYM/DRRG#17]
I think this line of codes are bit weired.
https://github.com/GXYM/DRRG/blob/master/util/misc.py#L223-L224

this
```
bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-2:]
bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[1]][:2]]
```
should be
```
bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-1:]
bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[0]][2:4]]
```

isn't it?

**Comment by GXYM** [comment]
> I think this line of codes are bit weired.
> https://github.com/GXYM/DRRG/blob/master/util/misc.py#L223-L224
> 
> this
> 
> ```
> bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-2:]
> bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[1]][:2]]
> ```
> 
> should be
> 
> ```
> bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-1:]
> bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[0]][2:4]]
> ```
> 
> isn't it?

no, what you The code you mentioned is from textsnake, please refer to https://github.com/princewang1994/TextSnake.pytorch/blob/master/util/misc.py. 
And we have partially fixed the bug，please refer to：
https://github.com/GXYM/DRRG/blob/ce63248ce44aac59821ccc1b405a64953aa91b51/util/misc.py#L197

this:
  bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-1]
   bottoms = [dist_list[bottom_idx][:2], dist_list[bottom_idx][2:4]]

**Comment by GXYM** [comment]
> > I think this line of codes are bit weired.
> > https://github.com/GXYM/DRRG/blob/master/util/misc.py#L223-L224
> > this
> > ```
> > bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-2:]
> > bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[1]][:2]]
> > ```
> > 
> > 
> > should be
> > ```
> > bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-1:]
> > bottoms = [dist_list[bottom_idx[0]][:2], dist_list[bottom_idx[0]][2:4]]
> > ```
> > 
> > 
> > isn't it?
> 
> no, what you The code you mentioned is from textsnake, please refer to https://github.com/princewang1994/TextSnake.pytorch/blob/master/util/misc.py.
> And we have partially fixed the bug，please refer to：
> 
> https://github.com/GXYM/DRRG/blob/ce63248ce44aac59821ccc1b405a64953aa91b51/util/misc.py#L197
> 
> this:
> if l1 > 2_l2 or l2 > 2_l1 or len1 == 0 or len2 == 0:
> for i in range(len(pts)):
> mid_point = (e[i] + e[(i + 1) % len(pts)]) / 2
> mid_list.append((i, (i + 1) % len(pts), mid_point))
> 
> ```
>                 for i in range(len(pts)):
>                     for j in range(len(pts)):
>                         s1, e1, mid1 = mid_list[i]
>                         s2, e2, mid2 = mid_list[j]
>                         dist = norm2(mid1 - mid2)
>                         dist_list.append((s1, e1, s2, e2, dist))
>                 bottom_idx = np.argsort([dist for s1, e1, s2, e2, dist in dist_list])[-1]
>                 bottoms = [dist_list[bottom_idx][:2], dist_list[bottom_idx][2:4]]
> ```

**Comment by jeong-tae** [comment]
Thx for the reply. I fixed code and it works well now.

I am not sure what is correct code. Hard to understand 100% but it works after fixing it.
